var/spack/repos/builtin/packages/node-js/package.py

```diff
@@ -138,6 +138,13 @@ class NodeJs(Package):
     # and https://github.com/nodejs/node/issues/53633
     patch("fix-broken-gcc12-pr53728.patch", when="@22.2:22.5")
 
+    # https://github.com/nodejs/node/issues/55596
+    # This patch is not sufficient, however, therefore
+    # add a conflict with this particular version of gcc
+    # until https://github.com/spack/spack/issues/48492 is resolved
+    patch("wasm-compiler-gcc11p2.patch", when="@21: %gcc@11.2")
```

Comment:
Should we apply the patch for all compilers when using an open version range? We tend to run into issue with these patches becoming impossible to apply when new versions are added.

Reply:
Not sure what you mean - I can remove the `%gcc` if that's what you want, or I can use a fixed range `@21:22?

Reply:
Yeah, removing the `%gcc` is best here to catch failing patches by default when testing with another compiler. But even closing the version range is appropriate here since the patch will fail on v23, ref https://github.com/nodejs/node/blob/v23.0.0/deps/v8/src/compiler/wasm-compiler.cc#L8679-L8686.

Reply:
Done, please check. I left the conflict with gcc@11.2 for all versions 21+, because 23 doesn't build with it either.